Description: release VADisplayContext at the end of vaTerminate()
 Otherwise it will result in invalid reading in va_TraceEnd()
Origin: upstream, http://cgit.freedesktop.org/libva/commit/?h=staging&id=d4988142a3f2256e38c5c5cdcdfc1b4f5f3c1ea9
Bug-Debian: https://bugs.debian.org/750429
Last-Update: 2014-06-03

diff --git a/va/va.c b/va/va.c
index 6cb17ef..0ba595b 100644
--- a/va/va.c
+++ b/va/va.c
@@ -515,15 +515,15 @@ VAStatus vaTerminate (
   free(old_ctx->vtable_vpp);
   old_ctx->vtable_vpp = NULL;
 
-  if (VA_STATUS_SUCCESS == vaStatus)
-      pDisplayContext->vaDestroy(pDisplayContext);
-
   VA_TRACE_LOG(va_TraceTerminate, dpy);
 
   va_TraceEnd(dpy);
 
   va_FoolEnd(dpy);
 
+  if (VA_STATUS_SUCCESS == vaStatus)
+      pDisplayContext->vaDestroy(pDisplayContext);
+
   return vaStatus;
 }
 
